sbUI and sbServer updated prototype

annazonked opened the pull request:
## Changes:

- Design changes on page 2 
 - Added page 4: guess and answer to simulated coin flip
 - Added page 5: thanks for participation
 - Changed overall button and background design

SebastianUniTue commented on main_app/illusions/sleepingbeauty/sbUI.R:
```diff
@@ -53,21 +154,69 @@ sbUI <- function(id) {
+          h3("Results"),
+          conditionalPanel(
+            condition = "input.lockButton > 0",
+            verbatimTextOutput("coinTossResult"),
```
should probably be:
```{r}
verbatimTextOutput(ns("coinTossResult")),
```

SebastianUniTue commented on main_app/illusions/sleepingbeauty/sbUI.R:
```diff
@@ -53,21 +154,69 @@ sbUI <- function(id) {
+          conditionalPanel(
+            condition = "input.lockButton > 0",
+            verbatimTextOutput("coinTossResult"),
+            verbatimTextOutput("guessComparison")
```
should probably be:
```{r}
verbatimTextOutput(ns("guessComparison"))
```

SebastianUniTue reviewed (changes requested):
Thank you for your code,
the problem you mentioned in the last sprint session is probably (I tested it with my changes and it works as intended now) due to you not using `ns("variable_name")` in your code. 

When you create a module you pass a string to your function (you called it `id`). 
When you call 
```{r}
sbUI("sleepingbeauty")
```
then `"sleepingbeauty"` is your `id`. the `ns` function in your code adds the `id` to the variable name. The variable name becomes 2 dim now if you will. The `id` defines a kind of namespace. This allows you to use the exact variable name in another module without them interfering. This creates modularity.

As for your other problem that Dominic mentioned (if you call 2 different `sbUI`s with different namespaces, they are sill interfering) right of the bat I belive it may be due to you using the variable names of, for example buttons without `ns` so that the other `ui` will access the same buttons.
If I'm right and this is the problem then it could be a problem. Another module without namespacing could then interfere with yours when it  uses the same variable names of the buttons.
This would be a problem that doesn't only affect you, but everyone who uses the `conditionalPanels` without adding `id`.

annazonked commented:
## To page 4: 
I managed to display all the desired outputs, but I'm getting the error message 'argument of length 0,' and I can't figure out what's causing it

Overall i added Text to page 2 and 8 and created a table showing the timetable of the experiment on page 2.

SebastianUniTue commented:
Thanks for your code,
you need to also need to use the `ns` function for your buttons.
```{r}
ns("userGuess")
```
The `module sever` expects his `input` to be 2 dimensional `(id,variable_name)`. The `id` is added by the `ns` function.
If you change this in your code the 3. text doesn't throw an error anymore and works as intended.

```{r}
output$guessComparison <- renderText({
      coinToss <- sample(c("Heads", "Tails"), 1)
      if (input$lockButton > 0) {
        if (input$userGuess == coinToss) {
          "You guessed right!"
        } else {
          "You guessed wrong!"
        }
      }
    })
```
As you use `lockButton` for the second one and this is also without `ns` this should also be your reason, why the second one isn't working. 
It could be that you have to change your more if you `ns` your button.

annazonked commented:
# Further changes: 
- Deleted page 4
- Changed the gif with a video

 Inserted two additional pages:
> Page 5: The question you get asked after you wake up
>Page 6: The end of the experiment, where you can proceed to further discussion of the SBProblem

# Style Changes
- renamed the buttons with the discussed conventions

this closes issue #22 because the problems that occured are resolved by the changed structure of the app.
